Increment targetSdkVersion to 27

audkar opened the pull request:
To increment version target to sdk 27 had to update other libs:
- build tools. Had dependencies:
  - gradle version 
- then support lib had dependencies
  - DebugDrawer had old version dependency to support. Updated to new version
  - 3rd party `material` lib had old version dependency to support. Replaced with official material lib.
  - Material dialogs library had old version dependency to support.

Seems that `location` runtime permissions kinda works. Would be nice to have better UX to notify user what happens when user refuses location permission.

I couldn't find place where is used `<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />`.  Maybe you know?

closes #72

kwatkins commented:
Write external storage was used when we wrote the ow logs to the sdcard, since @ebabel moved that to sqlite we may be able to drop that perm.

audkar commented:
`Export Ride` feature was writing file to external storage and then sharing that file. Moved it to `cache` folder and removed `WRITE_EXTERNAL_STORAGE` permission.
